liquibase-core/src/main/java/liquibase/util/CommonsFilenameUtils.java

```diff
@@ -0,0 +1,1684 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
```

Comment:
Bring in only what is needed here. Delete unused methods.

Also, make sure to bring in unit tests.